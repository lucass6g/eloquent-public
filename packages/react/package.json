{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
