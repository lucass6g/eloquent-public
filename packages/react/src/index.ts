export {Switch, Case, Default} from './components/conditional-switch'