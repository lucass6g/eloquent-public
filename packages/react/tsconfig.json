{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
