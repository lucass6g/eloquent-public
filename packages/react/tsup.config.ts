import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})