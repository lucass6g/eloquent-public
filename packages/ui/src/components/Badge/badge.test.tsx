import { render } from "@testing-library/react";
import { Badge } from ".";
import { describe, it, expect } from "vitest";

describe("Badge Component", () => {
  it("renders correctly with default variant", () => {
    const { container } = render(<Badge />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with the 'secondary' variant", () => {
    const { container } = render(<Badge variant="secondary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with the 'destructive' variant", () => {
    const { container } = render(<Badge variant="destructive" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with the 'outline' variant", () => {
    const { container } = render(<Badge variant="outline" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("applies additional className correctly", () => {
    const { container } = render(<Badge className="hover:bg-green-dark-600" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});