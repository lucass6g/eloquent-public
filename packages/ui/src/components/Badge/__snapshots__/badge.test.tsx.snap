// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge Component > applies additional className correctly 1`] = `
<div
  class="flex items-center rounded-md border px-2 py-1 text-sm transition-colors border-transparent bg-green-dark-500 text-neutral-100 hover:bg-green-dark-600"
/>
`;

exports[`Badge Component > renders correctly with default variant 1`] = `
<div
  class="flex items-center rounded-md border px-2 py-1 text-sm transition-colors border-transparent bg-green-dark-500 text-neutral-100"
/>
`;

exports[`Badge Component > renders correctly with the 'destructive' variant 1`] = `
<div
  class="flex items-center rounded-md border px-2 py-1 text-sm transition-colors border-transparent bg-red-600 text-neutral-100"
/>
`;

exports[`Badge Component > renders correctly with the 'outline' variant 1`] = `
<div
  class="flex items-center rounded-md border px-2 py-1 text-sm transition-colors border-neutral-300 text-neutral-800"
/>
`;

exports[`Badge Component > renders correctly with the 'secondary' variant 1`] = `
<div
  class="flex items-center rounded-md border px-2 py-1 text-sm transition-colors border-transparent bg-neutral-200 text-neutral-800"
/>
`;
